Не бейте сильно пишу первый раз в жизни :(

nim0y opened the pull request:
4 дня ужаса 
безнадеги 
5-6 удалений всего и вся
и этот "монстр франкенштейна" ~ "готов".
Сдаюсь.
С богом.)))
Забыл в FormRub  сделать 2ой метод на склонение слова Рубль.

nim0y committed:
Не бейте сильно пишу первый раз в жизни :(

AleksandrIlinskii commented on src/main/java/Calc.java:
```diff
@@ -0,0 +1,24 @@
+
+    public void finCountDown(int guestNumber) {
+        System.out.println("Добавлено:\n" + productNameC);
+        /*System.out.println(String.format(productNameC,System.lineSeparator()));*///хз нашел в интернетах,не разобрался
```
Если есть вопросы - всегда можно спросить у куратора.
Но комментарии лучше не пушить

nim0y replied:
Это для себя было , на случай если где то поломается.Спасибо!

AleksandrIlinskii commented on src/main/java/FormRub.java:
```diff
@@ -0,0 +1,19 @@
+            if ((int)(Calc.finalCost % 100 / 10) == 1){
+                return " рублей.";
+            }
+            switch ((int) (Calc.finalCost % 10)){
```
Совет на будущее: Если поставить курсор на switch, то можно увидеть фонарик желтый. Он подскажет улучшения кода

nim0y replied:
Тут два момента : 
1) Хотелось\показалось что Swich тут подходит.
2) при изменении выдает ошибки которые я не совсем понимаю как править.
![изображение_2023-04-19_080705750](https://user-images.githubusercontent.com/130760051/232972072-11b99850-a126-45c0-827a-4d4c9fe65998.png)

AleksandrIlinskii commented on src/main/java/Gests.java:
```diff
@@ -0,0 +1,30 @@
+
+
+
+
```
Слишком много пустых строк, лучше не стоит так делать

nim0y replied:
Принято убираю , в будущем постараюсь не оставлять.

AleksandrIlinskii commented on src/main/java/Gests.java:
```diff
@@ -0,0 +1,30 @@
+import java.util.Scanner;
+
+public class Gests {
```
Если предполагалось, что название класса переводится как "Гость", то тут опечатка

nim0y replied:
Да нет , я пока писал неслолько раз создавал похожие классы.
Крутил разные куски кода , в итоге что бы не путаться в блокноте получилось так.
Понимаю, буду внимательнее.

AleksandrIlinskii commented on src/main/java/Gests.java:
```diff
@@ -0,0 +1,30 @@
+
+public class Gests {
+    public static int guestNumber;
+    public static int Gest() {
```
И тут опечатка (см коммент выше)

AleksandrIlinskii replied:
Зачем делать возвращаемый тип у метода int, если ты его не используешь? Для таких целей есть void
<img width="459" alt="image" src="https://user-images.githubusercontent.com/129396632/232953544-a58d76ee-4d47-41fd-983f-9732878d0a0f.png">

AleksandrIlinskii commented on src/main/java/Main.java:
```diff
@@ -1,6 +1,16 @@
+
+
+
+
```
Тут тоже многовато строк

AleksandrIlinskii commented on src/main/java/Calc.java:
```diff
@@ -0,0 +1,24 @@
+public class Calc {
+    String productNameC = "";
```
Названия productName вполне достаточно для понимания зачем оно нужно

AleksandrIlinskii commented on src/main/java/FormRub.java:
```diff
@@ -0,0 +1,19 @@
+                case 4:
+                    return " рубля.";
+                default:
+                    return " рублей.";
```
Критическое замечание
Неверно выводится окончание слова рубль
![image](https://user-images.githubusercontent.com/129396632/232954125-19531659-1ab2-4028-91ba-94b5dfe40a85.png)

nim0y committed:
Спвсибо! Вы отлично работаете!

nim0y committed:
Перевел Gests в void, попробовал преобразовать Swich в 'if' с помощью "лампочки" (больше желтым не подсвечивает).
Незнаю можно ли делать commit после отправки ссылки на Практикум. Риск дело благородное.))

AleksandrIlinskii commented on src/main/java/FormRub.java:
```diff
@@ -4,31 +4,23 @@ public static String priceForm1() {
-            }
+        if ((int) (Calc.finalCost % 10) == 1) {
+            return " рубль.";
+        } else if ((int) (Calc.finalCost % 10) == 2 || (int) (Calc.finalCost % 10) == 3 || (int) (Calc.finalCost % 10) == 4) {
```
Как совет, ты можешь парой строчек выше завести переменную типа 
int lastDigit = (int) (Calc.finalCost % 10) и ее подставлять во все проверки твоих условий, а то сейчас получается, что ты считаешь одно и то же 4 раза. 
Это называется boilerplate code (код, который повторяется в нескольких местах). Очень легко так ошибиться
